tomviz/SaveLoadStateReaction.cxx

```diff
@@ -110,11 +114,34 @@ bool SaveLoadStateReaction::loadState(const QString& filename)
     }
   }
 
+  QDialog dialog(tomviz::mainWidget(), Qt::WindowStaysOnTopHint);
+  dialog.setModal(true);
+  dialog.setWindowTitle("Please wait...");
+
+  auto layout = new QHBoxLayout;
+  auto label =
+    new QLabel("Please wait... loading state file in progress", &dialog);
+  layout->addWidget(label);
+  dialog.setLayout(layout);
+  dialog.show();
+  // At least locally I need 2 processEvents calls for the dialog to show
+  // and appear non-blank.
+  QApplication::processEvents();
+  QApplication::processEvents();
   if (doc.isObject() && ModuleManager::instance().deserialize(
                           doc.object(), QFileInfo(filename).dir())) {
     RecentFilesMenu::pushStateFile(filename);
+    // Volume modules with 2D transfer functions need the histograms
+    // finsished before the user can interact.
+    while (HistogramManager::instance().hasPendingHistograms() ||
+           ModuleManager::instance().hasRunningOperators()) {
+      QApplication::processEvents();
+    }
+    dialog.hide();
```

Comment:
Think it would be cleaner to exec the dialog and the have signals produced by the HistogramManager and ModuleManager which we could connect to to close the dialog.

Reply:
Agreed.

Reply:
Yes, it seems that is the right way to go.  This way the dialog inconsistently shows up across machines. :(